### PR TITLE
Mvc content create json

### PR DESCRIPTION
**Pull request by tanmayranjan**

### CODE REVIEW

File: mvc-api/mvc-service/pom.xml [hunk 1]
```diff
@@ -108,6 +108,22 @@
...
+    <dependency>
+      <groupId>org.sunbird</groupId>
+      <artifactId>kafka-client</artifactId>
+      <version>1.0-SNAPSHOT</version>
+    </dependency>
+      <dependency>
+          <groupId>net.sf.opencsv</groupId>
+          <artifactId>opencsv</artifactId>
+          <version>2.3</version>
+      </dependency>
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/pom.xml]: Do we need this dependency.

File: mvc-api/mvc-service/pom.xml [hunk 2]
```diff
@@ -108,6 +108,22 @@
       <artifactId>logback-kafka-appender</artifactId>
       <version>0.2.0-RC2</version>
     </dependency>
+    <dependency>
+      <groupId>com.googlecode.json-simple</groupId>
+      <artifactId>json-simple</artifactId>
+      <version>1.1.1</version>
+      <scope>compile</scope>
+    </dependency>
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/pom.xml:116]: Do we still need this dependency?

File: mvc-api/mvc-service/app/controllers/SearchBaseController.scala [hunk 3]
```diff
@@ -129,4 +131,7 @@ abstract class SearchBaseController(protected val cc: ControllerComponents)(impl
         else if (null != searchRequest && null != searchRequest.getParams.getCid) searchRequest.put(TelemetryParams.ACTOR.name, searchRequest.getParams.getCid)
         else searchRequest.put(TelemetryParams.ACTOR.name, "learning.platform")
     }
+    /*protected def setVectorList(searchRequest: org.sunbird.common.dto.Request)(implicit playRequest: play.api.mvc.Request[AnyContent]) : Unit = {
+
+    }*/
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/app/controllers/SearchBaseController.scala]: Remove this.

File: mvc-api/mvc-service/app/managers/EventObjectProducer.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import org.sunbird.search.util.SearchConstants;
+
+import java.util.LinkedHashMap;
+import java.util.UUID;
+
+public class EventObjectProducer {
+    EventProducer eventProducer = new EventProducer();
+    public void addToEventObj(JSONObject content, String contentId)  {
+        // Add label
+        content.put("label","MVC");
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/app/managers/EventObjectProducer.java]: Are we still using `label` property?

File: mvc-api/mvc-service/app/managers/EventObjectProducer.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        ObjectMapper event = new ObjectMapper();
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/app/managers/EventObjectProducer.java]: Don't create this inside method. There is JsonUtils.java class. Please use that for any `Object` serialization and deserialization.

File: mvc-api/mvc-service/app/managers/EventObjectProducer.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            object.put("id", contentId);
+            LinkedHashMap<String,Object> context = (LinkedHashMap<String,Object>) eventObj.get("context");
+            context.put("channel", content.get("channel").toString());
+            LinkedHashMap<String,Object> edata = (LinkedHashMap<String,Object>) eventObj.get("edata");
+            edata.put("repository", SearchConstants.vidyadaanurl + contentId);
+            edata.put("metadata", content);
+            eventProducer.writeToKafka(eventObj.toString(), SearchConstants.mvctopic);
+        }
+        catch (Exception e) {
+            System.out.println("Exception in Eventobjectproducer " + e);
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/app/managers/EventObjectProducer.java]: The exception is suppressed here. We should throw it to the above level and return SERVER_ERROR (HTTP 500) or CLIENT_ERROR (HTTP 400) based on the error type.

File: mvc-api/search-core/src/main/java/org/sunbird/search/util/SearchConstants.java [hunk 7]
```diff
@@ -112,4 +112,18 @@
...
+    public static String contentreadapi = "api/content/v1/read/";
+    public static String autocreatejobevent = "{\"eid\":\"BE_JOB_REQUEST\",\"mid\":\"\",\"actor\":{\"id\":\"Auto Creator\",\"type\":\"System\"},\"context\":{\"pdata\":{\"ver\":\"1.0\",\"id\":\"org.ekstep.platform\"},\"channel\":\"\"},\"object\":{\"ver\":\"1.0\",\"id\":\"\"},\"edata\":{\"action\":\"auto-create\",\"iteration\":1,\"objectType\":\"Content\",\"repository\":\"\",\"metadata\":{}}}";
+    public static String mvctopic = "sunbirddock.auto.creation.job.request";
+    public static String mvcFailedtopic = "sunbirddock.auto.creation.job.failedevent";
+    public static String contentKeysObj = "{\"board\":\"board\",\"medium\":\"medium\",\"grade\":\"gradeLevel\",\"subject\":\"subject\",\"chapterconceptname\":\"level1Concept\",\"topicconceptname\":\"level2Concept\",\"subtopicconceptname\":\"level3Concept\",\"chaptername\":\"level1Name\",\"topicname\":\"level2Name\",\"subtopicname\":\"level3Name\",\"textbookname\":\"textbook_name\",\"contenturl\":\"sourceURL\"}";
+    public static String csvcolumndelimeter = "|";
+    public static String csvfiledelimeter =",";
+    public static int csvRowsLimit = 1000;
```
Comment (maheshkumargangula) [inline mvc-api/search-core/src/main/java/org/sunbird/search/util/SearchConstants.java]: Please check and delete all the unused code. After a few days, no one understands what are these variables.

**Review by maheshkumargangula — CHANGES_REQUESTED** [review]

File: mvc-api/mvc-service/pom.xml (hunk 1 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/pom.xml]: removing

File: mvc-api/mvc-service/pom.xml (hunk 2 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/pom.xml:116]: yes for json object

File: mvc-api/mvc-service/app/controllers/SearchBaseController.scala (hunk 3 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/app/controllers/SearchBaseController.scala]: okay

**Pull request closed without merging**